worker/pkg/benthos/sql/input_sql_raw.go

```diff
@@ -150,13 +146,9 @@ func (s *pooledInput) Connect(ctx context.Context) error {
 	s.dbMut.Lock()
 	defer s.dbMut.Unlock()
 
-	if s.db != nil {
```

Comment:
I removed this because Connect will retry until a non-error is returned. But if connect fails to retrieve the records, we will have set the DB, so the row input will be empty and Read will return with no input, which may be a source of bugs for us. What needs to happen is this needs to retry until Connect returns successfully.